### DISCUSSION TITLE
Documentation on Oauth

### DISCUSSION THREAD

**Discussion by rockerd1134** [TrueBrain/TrueWiki#209] [Q&A] [answered]
Hello,

I was wondering if there was an example of Oath or some other Single Sign On. I see the libraries make refence but I haven't figured out where to configure it. 

Thanks

**Comment by TrueBrain** [comment, marked as answer]
Currently the only two OAuth providers supported are GitHub and Gitlab. There is currently not a way, without modifying the code, to configure a generic OAuth provider.

There is some boilerplate to build from. See https://github.com/TrueBrain/TrueWiki/tree/main/truewiki/user for details.

The reason is that most OAuth providers need special settings and weirdness to actually be useful. So it depends a bit what you are looking for.

Was there any specific provider you were looking for?
For any public provider mentioned in https://github.com/klen/aioauth-client/blob/develop/aioauth_client/__init__.py (the library TrueWiki uses) adding support is relatively easy. If you are looking for some custom, that should be possible too.

For TrueWiki I add providers on demand, so they can be truly tested. So just let me know what you are looking for, and I can see what I can do for you :)

**Reply by rockerd1134** [reply]
Ok. Thanks for the information. I will attempt to get my gitlab to provide OAuth. 

MS Azure Oauth would be the specific provider. However my current production wiki is mediawiki with an MS Azure Simple Saml solution so I don't have the details for conversion. 

Thank you for you work on this project. I have been wanting to go to a git backed solution for a while but didn't want to install Gollum as the only Ruby app in my python ecosystem. So I am giving TrueWiki a try.

**Reply by TrueBrain** [reply]
> MS Azure Oauth would be the specific provider. However my current production wiki is mediawiki with an MS Azure Simple Saml solution so I don't have the details for conversion.

I might give adding Azure support a spin soon; shouldn't be too difficult I think. Would be an interesting test-case for me :)

> 
> Thank you for you work on this project. I have been wanting to go to a git backed solution for a while but didn't want to install Gollum as the only Ruby app in my python ecosystem. So I am giving TrueWiki a try.

Let me know how it goes. If you are missing anything or get stuck, feel free to ask for help!

**Reply by TrueBrain** [reply]
And as promised:

I just added Microsoft user support in TrueWiki [1.4.0](https://github.com/TrueBrain/TrueWiki/releases/tag/1.4.0). You have to create an Application Registration in your Azure Portal; here you can indicate what the user-scope of the application is (like: organization-only or "just any Microsoft account").

Two things worth mentioning:
- Microsoft doesn't seem to provide an URL where you can revoke permission to the application. 
- Microsoft doesn't supply anonymous emails like GitHub and Gitlab do, so commits are done based on the actual email address of the user.

Hopefully this helps you out a bit; if you need anything else, let me know!
